src/hypercore-file-logger.js

```diff
@@ -138,9 +138,12 @@ class HyperCoreFileLogger extends HyperCoreLogger {
     await super.start()
     this.watcher = chokidar.watch(this.pathlike)
 
+    let republish = this.republish
```

Comment:
would suggest to use different on ready than using this local variable, it would be running after the function is completed and keep it in memory
something like this would be better:
```js
    this.watcher.on('ready', () => { this.watcherReady = true })
    this.watcher.on('add', file => this.watchFile(file, {
      multiple: isGlob(this.pathlike),
      republish: this.republish || this.watcherReady
    }))
```